users/assets/js/script.js

```diff
@@ -0,0 +1,303 @@
+}
+
+
+function validateForm() {
+    var username = document.getElementById("username").value;
+    var name = document.getElementById("name").value;
+    var email = document.getElementById("email").value;
+    var password = document.getElementById("password").value;
+    var confirm_password = document.getElementById("confirm_password").value;
+
+    if (username === "" || name === "" || email === "") {        
+        document.querySelectorAll("#username_error,#name_error,#email_error").forEach(function(element) {
+            element.innerText = "All fields are required.";        
+        });
+        return false;        
+    }
+
+    var isPasswordValid = validatePassword(password, confirm_password);
+    if (!isPasswordValid) {
+        return false;
+    }
+
+    var isPasswordSame = confirmPassword(password, confirm_password);
+    if (!isPasswordSame) {
+        return false;
+    }
+    
+    var isEmailValid = validateEmail(email);    
+    if (!isEmailValid) {
+        return false;
+    }
+
+    // Clear the error message if everything is valid
+    document.querySelector(".error_message").innerText = "";
+    return true; // If all validations pass, the form can be submitted
+}
+
+function validateFormEdit() {
+    // Fetch only the necessary fields for edit form validation
+    var password = document.getElementById("password").value;
+    var confirm_password = document.getElementById("confirm_password").value;
+    var email = document.getElementById("email").value;
+
+    // Perform validation only if the fields contain values
+    if (password) {
+        var isPasswordValid = validatePassword(password, confirm_password);
+        if (!isPasswordValid) {
+            return false;
+        }
+    }
+
+    if (confirm_password) {
+        var isPasswordSame = confirmPassword(password, confirm_password);
+        if (!isPasswordSame) {
+            return false;
+        }
+    }
+
+    if (email) {
+        var isEmailValid = validateEmail(email);
+        if (!isEmailValid) {
+            return false;
+        }
+    }
+
+    // Clear the error message if everything is valid
+    document.querySelector(".error_message").innerText = "";
+    return true; // If all validations pass, the form can be submitted
+}
+
+function removeImagePreview(fileInput,imgPreviewWrapper,imgPreview,imgFileName,imgFileNameError) {
+    fileInput.value = ""; 
+    imgPreview.src = "#"; 
+    imgPreviewWrapper.style.display = "none";
+    imgFileName.innerText = "Choose Profile Picture";
+    imgFileNameError.innerText = "No file selected"; 
+}
+
+        
+
+        jQuery(document).on("click", "#edit_my_details", function() {
+            jQuery('#edit_user_form').toggle();
+            var editFormOffset = jQuery("#edit_user_form").offset().top;
+            jQuery("html, body").animate({ scrollTop: editFormOffset }, 1000);
+        });
+
+
+        function loadUserTable() {
+            $.ajax({
```

Comment:
can we hit this function only once when every other field has been validated in the front end, till then we can avid multiple AJAX calls by disabling the submit button

Reply:
This is fixed